daemon/api_systems.go

```diff
@@ -104,19 +105,45 @@ func getAllSystems(c *Command, r *http.Request, user *auth.UserState) Response {
 }
 
 // wrapped for unit tests
-var deviceManagerSystemAndGadgetInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, error) {
-	return dm.SystemAndGadgetInfo(systemLabel)
+var deviceManagerSystemAndGadgetAndEncryptionInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, *devicestate.EncryptionSupportInfo, error) {
+	return dm.SystemAndGadgetAndEncryptionInfo(systemLabel)
+}
+
+func storageEncryption(encInfo *devicestate.EncryptionSupportInfo) *client.StorageEncryption {
+	if encInfo.Disabled {
+		return &client.StorageEncryption{
+			Support: client.StorageEncryptionSupportDisabled,
+		}
```

Comment:
this path is not tested

Reply:
I added a bunch more table driven testcases now.